tests/integration/security/file_mounted_certs/main_test.go

```diff
@@ -0,0 +1,204 @@
+// +build integ
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package filemountedcerts
+
+import (
+	"context"
+
+	"io/ioutil"
+	"path"
+	"strings"
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	inst istio.Instance
+)
+
+const (
+	ProxyMetadataJson = `
+	{
+      "JWT_POLICY": "third-party-jwt",
+      "PILOT_CERT_PROVIDER": "mycopki",
+      "FILE_MOUNTED_CERTS": "true",
+      "ISTIO_META_TLS_CLIENT_CERT_CHAIN": "/client-certs/cert.pem",
+      "ISTIO_META_TLS_CLIENT_KEY": "/client-certs/key.pem",
+      "ISTIO_META_TLS_CLIENT_ROOT_CERT": "/client-certs/ca.pem",
+      "ISTIO_META_TLS_SERVER_CERT_CHAIN": "/server-certs/cert.pem",
+      "ISTIO_META_TLS_SERVER_KEY": "/server-certs/key.pem",
+      "ISTIO_META_TLS_SERVER_ROOT_CERT": "/server-certs/ca.pem",
+      "PROXY_XDS_VIA_AGENT": "false",
+      "ISTIO_METAJSON_METRICS_INCLUSIONS": '{\"sidecar.istio.io/statsInclusionPrefixes\": \"access_log_file,cluster,cluster_manager,control_plane,http,http2,http_mixer_filter,listener,listener_manager,redis,runtime,server,stats,tcp,tcp_mixer_filter,tracing\"}',
+      "ISTIO_META_DNS_CAPTURE": "false",
+	}
+`
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		Label(label.CustomSetup).
+
+		// SDS requires Kubernetes 1.13
+		RequireEnvironmentVersion("1.13").
+		Label("CustomSetup").
+		Setup(istio.Setup(&inst, setupConfig, CreateCustomIstiodSecret)).
+		Run()
+}
+
+
+func setupConfig(_ resource.Context, cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+
+	cfg.ControlPlaneValues = `
+components:
+  egressGateways:
+  - enabled: false
+    name: istio-egressgateway
+  ingressGateways:
+  - enabled: false
+    name: istio-ingressgateway
+  pilot:
+    enabled: true
+    k8s:
+      overlays:
+        - kind: Deployment
+          name: istiod
+          patches:
+            - path: spec.template.spec.containers.[name:discovery].args
```

Comment:
nit: does the overlay allow appending or only replacing? If appending we should prefer this

Reply:
I tried a few variations and the following seems to work:
```
            - path: spec.template.spec.containers.[name:discovery].args[1001]
              value: "--caCertFile=/server-certs/ca.pem"
            - path: spec.template.spec.containers.[name:discovery].args[1002]
              value: "--tlsCertFile=/server-certs/cert.pem"
            - path: spec.template.spec.containers.[name:discovery].args[1003]
              value: "--tlsKeyFile=server-certs/key.pem"

```

I'll update the code.